cmd/scollector/main.go

```diff
@@ -403,6 +403,17 @@ func parseHost(host string) (*url.URL, error) {
 	return u, nil
 }
 
+func hideUrlCredentials(u *url.URL) *url.URL {
+	// Copy original url, replace credentials, e. g. for logging
+	if u.User != nil {
+		u2 := new(url.URL)
```

Comment:
This should only do anything if the user is already set. Don't want to add xxx if there is no user to begin with.

Reply:
I am not sure I understand. That's why there is `if u.User != nil` check -- only when a User was set in the original URL, a copy will be made, with the xxx instead of the original credentials. So unless I am overlooking something, this already does what you ask for?

Reply:
wow, I spaced that one. Yeah, you're right.